internal/api/api.go

```diff
@@ -2,22 +2,75 @@ package api
 
 import (
 	"context"
+	"crypto/tls"
+	"crypto/x509"
 	"fmt"
+	"net/url"
+	"strings"
+	"time"
 
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/timeout"
 	"github.com/intelops/go-common/logging"
 	"github.com/intelops/vault-cred/config"
 	"github.com/intelops/vault-cred/internal/client"
 	"github.com/intelops/vault-cred/proto/pb/vaultcredpb"
+	"github.com/kelseyhightower/envconfig"
+
 	"github.com/pkg/errors"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+	"google.golang.org/grpc/credentials/insecure"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	kadAppRolePrefix = "kad-approle-"
 )
 
 type VaultCredServ struct {
 	vaultcredpb.UnimplementedVaultCredServer
-	conf config.VaultEnv
-	log  logging.Logger
+	conf        config.VaultEnv
+	log         logging.Logger
+	connection  *grpc.ClientConn
+	cfg         *Config
+	vaultClient vaultcredpb.VaultCredClient
+}
+type Config struct {
+	Address     string
+	CaCert      string
+	Cert        string
+	Key         string
+	ServicName  string
+	AuthEnabled bool
+	TlsEnabled  bool
+}
+
+func FetchConfig() (Config, error) {
+	cfg := Config{}
+	err := envconfig.Process("", &cfg)
+	return cfg, err
+}
+
+func NewVaultClient(log logging.Logger, cfg *Config) (*VaultCredServ, error) {
+	log.Infof("connecting to VaultClient %s", cfg.Address)
+
+	conn, err := getConnection(cfg)
+	if err != nil {
+		return nil, errors.WithMessage(err, "failed to connect to agent")
+	}
+
+	vaultClient := vaultcredpb.NewVaultCredClient(conn)
+
+	return &VaultCredServ{
+		log:         log,
+		cfg:         cfg,
+		connection:  conn,
+		vaultClient: vaultClient,
+	}, nil
 }
```

Comment:
should not create client within same server. This grpc call is not needed, function is is within same process. Make just function call